src/i_midimusic.c

```diff
@@ -1310,19 +1311,40 @@ static void GetDevices(void)
         midiout = rtmidi_out_create_default();
     }
 
+    if (CheckError("GetDevices"))
+    {
+        return;
+    }
+
+    enum RtMidiApi current_api = rtmidi_out_get_current_api(midiout);
+
+    if (CheckError("GetDevices"))
+    {
+        return;
+    }
+
     unsigned int num_devs = rtmidi_get_port_count(midiout);
 
     for (int i = 0; i < num_devs; ++i)
     {
-        char *s;
         int len;
         rtmidi_get_port_name(midiout, i, NULL, &len);
-        s = malloc(len);
-        rtmidi_get_port_name(midiout, i, s, &len);
-        array_push(ports, s);
-    }
+        char *name = malloc(len);
+        rtmidi_get_port_name(midiout, i, name, &len);
 
-    CheckError("GetDevices");
+        if (current_api == RTMIDI_API_LINUX_ALSA)
+        {
+            char *p = strchr(name, ':');
```

Comment:
I think this breaks when the name is not unique. A simple example is Timidity which has a few ports open by default. If port 2 is selected, the wrong port is used the next time Woof is launched (always goes back to the first found port I think).
```
$ aplaymidi -l
 Port    Client name                      Port name
 14:0    Midi Through                     Midi Through Port-0
128:0    TiMidity                         TiMidity port 0
128:1    TiMidity                         TiMidity port 1
128:2    TiMidity                         TiMidity port 2
128:3    TiMidity                         TiMidity port 3
```

Reply:
We can save the full port name in config, but display only the client name and port number in the menu.

Reply:
Fixed by switching to PortMidi.